.github/workflows/scripts/e2e/gmc_install.sh

```diff
@@ -0,0 +1,147 @@
+    # Copy manifest into gmc
+    mkdir -p $(pwd)/config/manifests
+    cp $(dirname $(pwd))/manifests/ChatQnA/*.yaml -p $(pwd)/config/manifests/
+    cp $(dirname $(pwd))/manifests/DocSum/gaudi/docsum_gaudi_llm.yaml -p $(pwd)/config/manifests/
+}
+
+function init_gmc() {
+    # copy manifests
+    copy_manifests_${PLATFORM}
+
+    # replace tag with for the gmc-router and gmc-manager image
+    sed -i "s|opea/\(.*\):latest|opea/\1:$VERSION|g" $(pwd)/config/gmcrouter/gmc-router.yaml
+    sed -i "s|opea/\(.*\):latest|opea/\1:$VERSION|g" $(pwd)/config/manager/gmc-manager.yaml
+    # replace the pull policy "IfNotPresent" with "Always"
+    sed -i "s#pullPolicy: IfNotPresent#pullPolicy: Always#g" $(pwd)/config/gmcrouter/gmc-router.yaml
+    sed -i "s#pullPolicy: IfNotPresent#pullPolicy: Always#g" $(pwd)/config/manager/gmc-manager.yaml
+
+    cp $(pwd)/config/gmcrouter/gmc-router.yaml -p $(pwd)/config/manifests/
+
+    # replace namespace for gmc-router and gmc-manager
+    sed -i "s|namespace: system|namespace: $SYSTEM_NAMESPACE|g"  $(pwd)/config/manager/gmc-manager.yaml
+    sed -i "s|namespace: system|namespace: $SYSTEM_NAMESPACE|g"  $(pwd)/config/rbac/gmc-manager-rbac.yaml
+    sed -i "s|name: system|name: $SYSTEM_NAMESPACE|g" $(pwd)/config/rbac/gmc-manager-rbac.yaml
+
+    # replace the mount dir "path: /mnt/model" with "path: $CHART_MOUNT"
+    find . -name '*.yaml' -type f -exec sed -i "s#path: /mnt/models#path: $MOUNT_DIR#g" {} \;
+    # replace the repository "image: opea/*" with "image: ${IMAGE_REPO}opea/"
+    find . -name '*.yaml' -type f -exec sed -i "s#image: opea/*#image: ${IMAGE_REPO}opea/#g" {} \;
+    find . -name '*.yaml' -type f -exec sed -i "s#image: \"opea/*#image: \"${IMAGE_REPO}opea/#g" {} \;
+    # set huggingface token
+    # find . -name '*.yaml' -type f -exec sed -i "s#insert-your-huggingface-token-here#$(cat $TOKEN_DIR)#g" {} \;
+    find . -name '*.yaml' -type f -exec sed -i "s#insert-your-huggingface-token-here#$(cat $TOKEN_DIR)#g" {} \;
+}
+
+function cleanup_gmc() {
+    echo "clean up microservice-connector"
+    if kubectl get namespace $SYSTEM_NAMESPACE > /dev/null 2>&1; then
+        echo "Deleting namespace: $SYSTEM_NAMESPACE"
+        kubectl delete namespace "$SYSTEM_NAMESPACE"
+        kubectl delete crd gmconnectors.gmc.opea.io
+    else
+        echo "Namespace $SYSTEM_NAMESPACE does not exist"
+    fi
+}
+
+function wait_until_pod_ready() {
+    echo "Waiting for the $1 to be ready..."
+    max_retries=30
+    retry_count=0
+    while ! is_pod_ready $2 $3; do
+        if [ $retry_count -ge $max_retries ]; then
+            echo "$1 is not ready after waiting for a significant amount of time"
+            get_gmc_controller_logs
+            exit 1
+        fi
+        echo "$1 is not ready yet. Retrying in 30 seconds..."
+        sleep 30
+        output=$(kubectl get pods -n $2)
+        echo $output
+        retry_count=$((retry_count + 1))
+    done
+}
+
+function is_pod_ready() {
```

Comment:
we should make this function more generic and move it to a common scripts to be used by various test cases. But it can be done later